schema-registry/client/src/main/java/com/hortonworks/registries/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -1253,6 +1253,7 @@ public String run() {
                                       String.class,
                                       "URL of schema registry to which this client connects to. For ex: http://localhost:9090/api/v1",
                                       "http://localhost:9090/api/v1",
+                                      ConfigEntry.StringConverter.get(),
```

Comment:
The type of the ConfigEntry is already defined. Can we implicitly choose the converter instead of passing it in constructor?

Reply:
Is it possible? I mean, in an elegant way.
Once a `ConfigEntry<T>` is created, we know its type parameter `T`, so in case, for instance,  of a `String` a value of type `Converter<String>` can be assigned to a variable typed `Converter<? extends String>`.

But what about the general case? We should construct an expression whose result can be either `Converter<Integer>` or `Converter<String>` and assign to (or return as) something typed `Converter<? extends something>`. The common ancestor of `String` and `Integer` is only `Object`. In this case we would have a `Converter<Object>`, which a solution to this problem - but not a useful one. I think, it is cleaner to stick to the way as, for example, a `Validator` is chosen. All the information we need is present at compile time.

Reply:
You can refer this method (https://github.com/apache/kafka/blob/eaccb9292984455a2978a635a224fb51f3b051b9/clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java#L633) which parses the value based on it's type.